### PR TITLE
cpu/esp32: update documentation for ESP32-C3

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR is a split-off from PR #17844, which updates the documentation for ESP32-C3.

### Testing procedure

Green CI without compilation test since only `*.txt` files are changed.

### CODE REVIEW

File: cpu/esp32/doc.txt [hunk 1]
```diff
@@ -166,7 +166,7 @@ either based on
 - Tensilica Xtensa 32-bit LX7 microprocessor (ESP32-S2, ESP32-S3), or
 - 32-bit RISC-V CPU (ESP32-C3, ESP32-H2).
 
-At the moment, ESP32 variant is supported by RIOT-OS.
+At the moment, ESP32 and ESP32-C3 variants (families) are supported by RIOT-OS.
```
Comment (benpicco) [inline cpu/esp32/doc.txt:169]: Now this is a lie until #18345 is merged :wink:

**Review by benpicco — APPROVED** [review]
No need to split off non-logic changes, but maybe this is then a good motivation to merge #18345 soon :smile:

**Comment by gschorcht** [comment]
> No need to split off non-logic changes, but maybe this is then a good motivation to merge #18345 soon

I simply tried to reduce the size of the core PR as much as possible.

File: cpu/esp32/doc.txt (hunk 1 above)
Comment (gschorcht) [reply cpu/esp32/doc.txt:169]: Yes, but I think this is acceptable in a development branch for a short time. Initially I thought we could merge PR #18345 first and then this PR until I noticed that a reference to the documentation is used in the header files of PR #18345 :worried:

**Pull request merged**